op-node/rollup/driver/state.go

```diff
@@ -212,8 +215,21 @@ func (s *Driver) eventLoop() {
 		// And avoid sequencing if the derivation pipeline indicates the engine is not ready.
 		if s.driverConfig.SequencerEnabled && !s.driverConfig.SequencerStopped &&
 			s.l1State.L1Head() != (eth.L1BlockRef{}) && s.derivation.EngineReady() {
-			// update sequencer time if the head changed
-			if s.sequencer.BuildingOnto().ID() != s.derivation.UnsafeL2Head().ID() {
+			if s.driverConfig.SequencerMaxSafeLag > 0 && s.derivation.SafeL2Head().Number+s.driverConfig.SequencerMaxSafeLag <= s.derivation.UnsafeL2Head().Number {
+				// If the safe head has fallen behind by a significant number of blocks, delay creating new blocks
+				// until the safe lag is below SequencerMaxSafeLag.
+				s.log.Warn(
+					"Delay creating new block since safe lag exceeds limit",
+					"safe.Number", s.derivation.SafeL2Head().Number,
+					"unsafe.Number", s.derivation.UnsafeL2Head().Number,
+				)
+				blockTime := time.Duration(s.config.BlockTime) * time.Second
+				resetSequencerCh(blockTime)
```

Comment:
I would just set it to `nil` like when sequencing is disabled. It'll get re-enabled the first loop iteration where the safe head and unsafe head difference is not as big anymore, and `planSequencerAction()` is called.
That also makes the change smaller, as you do not have to call `resetSequencerCh(time)` explicitly.

Reply:
Excellent idea! I will definitely try that.